src/snowflake/connector/connection.py

```diff
@@ -1433,19 +1434,30 @@ def get_query_status_throw_if_error(self, sf_qid: str) -> QueryStatus:
         """
         status, status_resp = self._get_query_status(sf_qid)
         queries = status_resp["data"]["queries"]
+        try:
+            found_query = queries[0]
+        except IndexError:
+            Error.errorhandler_wrapper(
```

Comment:
would this be a breaking change to user?
we wrap an `IndexError` into a `ProgrammingError`, user code catching `IndexError` previously won't work now.

Reply:
Yes, I was okay waiting for this to be pre-announced, but maybe I should just split this PR into 2

Reply:
Split into #1159

Reply:
this change looks totally reasonable to me.
However, errors are part of the public apis, I'm just curious that when we make public api changes, is there any procedure (e.g. approval) we need to go through or we just make the change and announce it.

Reply:
@sfc-gh-mkeller What do we usually do for bug-fixing breaking change?

Reply:
thanks for the info!